lib/spack/spack/test/stage.py

```diff
@@ -18,27 +21,99 @@
 from spack.resource import Resource
 from spack.stage import Stage, StageComposite, ResourceStage
 
+_archive_base = 'test-files'
+_archive_fn = '%s.tar.gz' % _archive_base
+_extra_fn = 'extra.sh'
+_hidden_fn = '.hidden'
+_readme_fn = 'README.txt'
 
-def check_expand_archive(stage, stage_name, mock_archive):
+_extra_contents = '#!/bin/sh\n'
+_hidden_contents = ''
+_readme_contents = 'hello world!\n'
+
+# TODO: Replace the following with an enum once guarantee supported (or
+# include enum34 for python versions < 3.4.
+_include_readme = 1
+_include_hidden = 2
+_include_extra = 3
+
+
+def check_expand_archive(stage, stage_name, expected_file_list):
+    """Ensure the expanded archive directory contains the expected files."""
+    # Expect expanded stage area to look like the following for a well-behaved:
+    # tarball:
+    #
+    # TMPDIR/                temp stage dir
+    #     src/               well-known stage source directory
+    #         <_readme_fn>   Optional test_readme (contains _readme_contents)
+    #     <_hidden_fn>       Optional hidden file (contains _hidden_contents)
+    #     <_archive_fn>      archive_url = file:///path/to/<_archive_fn>
+    #
+    # and like the following for an exploding tarball:
+    #
+    # TMPDIR/                  temp stage dir
+    #     src/                 well-known stage source directory
+    #         <archive_name>/  archive dir
+    #             <_readme_fn> test_readme (contains _readme_contents)
+    #         <_extra_fn>      test_extra file (contains _extra_contents)
+    #     <_archive_fn>        archive_url = file:///path/to/<_archive_fn>
+    #
```

Comment:
Given that the new names now apply to more than one test, I'd move the descriptive comment up with the module-scope vars.

Reply:
Will do.